src/Relation/Binary/PropositionalEquality.agda

```diff
@@ -104,15 +104,17 @@ module _ (_≟_ : DecidableEquality A) {x y : A} where
   ≢-≟-identity = dec-no (x ≟ y)
 
 
+------------------------------------------------------------------------
+-- Inspect
 
+-- Inspect can be used when you want to pattern match on the result r
+-- of some expression e, and you also need to "remember" that r ≡ e.
```

Comment:
One of the (many? subtle!)  peculiarities of `inspect` us that it does rely of `e` being of the form `f a` and requires you to identify those pieces separately. That is, there seems no way to fold together `with e` with an invocation of `inspect e` (the latter makes no sense *per se*) in a single statement form. 

Unless one perhaps writes directly 
```agda
with inspect {.(f a)} f a
... | eq = ...
```
but even then the explicit dot-patterned term is not bound to a(n independent) variable subsequently available to the matcher. 

So even the pragmatic of this notation are both annoying, and hard to explain well...